zbus/src/match_rule_builder.rs

```diff
@@ -200,12 +200,73 @@ impl<'m> MatchRuleBuilder<'m> {
     }
 
     /// Set 0th argument's namespace.
+    ///
+    /// This function is deprecated because the choice of `InterfaceName` was too restrictive.
+    #[deprecated = "use arg0ns instead"]
     pub fn arg0namespace<I>(mut self, namespace: I) -> Result<Self>
     where
         I: TryInto<InterfaceName<'m>>,
         I::Error: Into<Error>,
     {
-        self.0.arg0namespace = Some(namespace.try_into().map_err(Into::into)?);
+        let namespace = namespace.try_into().map_err(Into::into)?;
+        self.0.arg0namespace = Some(namespace.clone());
+        self.0.arg0ns = Some(namespace.into());
+
+        Ok(self)
+    }
+
+    /// Set 0th argument's namespace.
+    ///
+    /// The namespace be a valid bus name or a valid element of a bus name. For more information,
+    /// see [the spec](https://dbus.freedesktop.org/doc/dbus-specification.html#message-protocol-names-bus).
+    ///
+    /// Eamples of valid namespaces:
+    /// - `org.mpris.MediaPlayer2`
+    /// - `org`
+    /// - `:org`
+    /// - `:1org`
+    ///
+    /// Eamples of invalid namespaces:
+    /// - `org.`
+    /// - `.org`
+    /// - `1org`
+    /// - `.`
+    /// - `org..freedesktop`
```

Comment:
Nice! Even better if we convert this into a test case. See the `ObjectPath` docs.

Reply:
You mean a doctest?

Reply:
`unwrap_err` requires that `MatchRuleBuilder` implements `Debug`. Is it OK if I add `derive(Debug)` (is a separate commit)?

Reply:
> You mean a doctest?

yes.

> `unwrap_err` requires that `MatchRuleBuilder` implements `Debug`. Is it OK if I add `derive(Debug)` (in a separate commit)?

Sure thing. It's probably a desired change on its own anyway.